File: Tps/Tp1/Palindromo.c
#include <stdio.h>
#include <string.h>

#define MAX 1000

int comparaString(char *palavra) 
{
    return strcmp(palavra, "FIM");
}

int ehPalindromo(char palavra[], int inicio, int fim) 
{
    if (inicio >= fim) 
    {
        return 1;
    }
    if (palavra[inicio] != palavra[fim]) 
    {
        return 0;
    }
    return ehPalindromo(palavra, inicio + 1, fim - 1);
}

int main() 
{
    char palavra[MAX];
    int resultado = 1;

    while (resultado != 0) 
    {
        fgets(palavra, MAX, stdin);
        palavra [strcspn(palavra, "\n")] = '\0';
        
        resultado = comparaString(palavra);

        if (comparaString(palavra) == 0) 
        {
            break;
        }

        int tamanho = strlen(palavra);

        if (ehPalindromo(palavra, 0, tamanho - 1)) 
        {
            printf("SIM\n");
        } 
        else 
        {
            printf("NAO\n");
        }
    }

    return 0;
}

File: Tps/Tp1/Palindromo.java
import java.util.*;

class Palindromo
    {
            
        public static int comparaString ( String palavra)
        {
           String str1 = "FIM";
           return palavra.compareTo(str1);
        }
        
        public static boolean ehPalindromo ( String palavra) 
        {
            int tamanho = palavra.length();
            for ( int i = 0; i < tamanho / 2; i ++)
                {
                    if ( palavra.charAt(i) != palavra.charAt(tamanho - 1 - i))
                    {
                        return false;
                    }
                }
            return true; 
        }

        public static void print( )
        {
            Scanner sc = new Scanner(System.in);
            int resultado = 1;
            while ( resultado != 0 )
                {
                    String palavra;
                    // System.out.println("Digite uma palavra: ");
                    palavra = sc.nextLine();
                    resultado = comparaString(palavra);

                    if ( resultado != 0)
                        {
                           if (ehPalindromo(palavra)) 
                           {
                               System.out.println("SIM");
                           } 
                           else 
                           {
                            System.out.println("NAO");
                           }  
                        }
                    
                }
                sc.close();
            }
        
        public static void main ( String [] args )
        {
            print();
        }
    }